app/src/main/java/io/appium/settings/Settings.java

```diff
@@ -48,7 +51,8 @@ public void onCreate(Bundle savedInstanceState) {
         setContentView(R.layout.main);
         Log.d(TAG, "Entering the app");
 
-        LocationTracker.getInstance().start(this);
+        FusedLocationProviderClient fusedLocationClient = LocationServices.getFusedLocationProviderClient(this);
```

Comment:
what is the reason to move fused location client creation here?

Reply:
I notice you use it in LocationService, but it is likely a mocked service. As I am not so faimiliar the codes, so I just use this here for quick realization. 
Another solution is to change current _LocationService_ to _MockLocationService_ and add real _LocationService_ to serve here to avoid use _FusedLocationProviderClient_ in _Settings_.
What's the condition of mocklocationservice used?

Reply:
I would prefer to use a common approach everywhere. If we create location service instances outside of the service then we have to do it for both coarse and fine locators. Otherwise there is probably no need to pass `this` instance to the `start` method.